### PR TITLE
Measure the user responsiveness to notifications over time

### PR DESCRIPTION
**Pull request by matthiasmullie**
https://phabricator.wikimedia.org/T108208

### CODE REVIEW

File: graphs/enwiki_echo_distribution_of_response_time.json [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+          },
+          "dimension": "x",
+          "orient": "bottom"
+        }
+      },
+      {
+        "nodeType": "axis",
+        "disabled": false,
+        "options": {
+          "tickFormat": "YY",
```
Comment (matthiasmullie) [inline graphs/enwiki_echo_distribution_of_response_time.json]: 'YY' seems to suggest date (year)-based, but this left axis should just be numbers. Not sure what that format should be.

File: graphs/enwiki_echo_distribution_of_response_time.json [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+          },
+          "dimension": "y"
+        }
+      },
+      {
+        "nodeType": "legend",
+        "disabled": false,
+        "options": {
+          "dateFormat": "MMM YYYY",
+          "valueFormat": ",.2s"
```
Comment (matthiasmullie) [inline graphs/enwiki_echo_distribution_of_response_time.json]: What format is this exactly?

File: graphs/enwiki_echo_distribution_of_unread_notifications.json [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  "notes": "",
+  "root": {
+    "nodeType": "canvas",
+    "disabled": false,
+    "children": [
+      {
+        "nodeType": "axis",
+        "disabled": false,
+        "options": {
+          "tickFormat": "YY",
```
Comment (matthiasmullie) [inline graphs/enwiki_echo_distribution_of_unread_notifications.json]: This axis should be a percentage.

**Comment by milimetric** [comment]
:( I'm sorry you had to do this horrible limn config work.  It looks like you're just adding enwiki graphs, is that right?  The data you're creating will be for several different dbs, all in different files.  There's more work involved if you want to hook up the graphs in this awful manual way, but you can also use the shortcut of just linking to the file on datasets.wikimedia.org.  Here's how mobile used that in their dashboard: https://github.com/wikimedia/analytics-limn-mobile-data/blob/master/dashboards/reportcard.json

In your case, you could make 3 new tabs, one for each SQL file you have.  And in each of those tabs, link to where the datafile for each wiki will end up.  This is what it would look like if you did that (the change to your dashboard):

```
"tabs": [
    ...
    ... (the existing tabs config)
    ...
    {
        "name": "Distribution of Response Time",
        "graph_ids": [
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/distribution_of_response_time/enwiki.tsv",
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/distribution_of_response_time/dewiki.tsv",
            ...
        ]
    },{
        "name": "Distribution of Unread Notifications",
        "graph_ids": [
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/distribution_of_unread_notifications/enwiki.tsv",
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/distribution_of_unread_notifications/dewiki.tsv",
            ...
        ]
    },{
        "name": "Monthly Production and Consumption of Notifications",
        "graph_ids": [
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/monthly_production_and_consumption_of_notifications/enwiki.tsv",
            "http://datasets.wikimedia.org/limn-public-data/metrics/echo/monthly_production_and_consumption_of_notifications/dewiki.tsv",
            ...
        ]
    }
]
```

This way, you wouldn't have to change anything else and you get some default graphs based on all the columns in those files.  We can talk more if you want to have precise control of how the graphs show.

**Comment by matthiasmullie** [comment]
Instead of creating new tabs for these charts, I've added them to the existing tabs. Does that also work?

**Comment by milimetric** [comment]
That works, but with the current version of the code the name will just be
"enwiki" (it just uses the last part of the name).  Hm, I should fix that
and allow a name to be passed, are you ok waiting a little for that?  I'll
try to get it done today.

**Comment by matthiasmullie** [comment]
Sure, awesome, no rush! :)

**Comment by milimetric** [comment]
Ok, @matthiasmullie, this was deployed.  So now instead of just the URL to the file, you can do something like:

"http://datasets.wikimedia.org/limn-public-data/metrics/echo/distribution_of_unread_notifications/enwiki.tsv?name=Distribution of Unread Notifications"

**Comment by matthiasmullie** [comment]
Perfect, thanks!

**Pull request merged**